instructor/dsl/maybe.py

```diff
@@ -53,23 +53,22 @@ def __bool__(self):
     Returns:
         MaybeModel (Type[BaseModel]): A new Pydantic model that includes fields for `result`, `error`, and `message`.
     """
-
-    fields = {
-        "result": (
+    return create_model(
```

Comment:
We would have to use a type-dict to unpack in a type-safe manner, so it's just going to be less verbose to directly pass everything in.